cadquery/cq.py

```diff
@@ -3800,6 +3800,62 @@ def sphere(
         else:
             return self.union(spheres, clean=clean)
 
+    def cone(
+        self: T,
+        height: float,
+        radius: float = 0,
+        radius1: float = None,
+        radius2: float = None,
+        direct: Vector = Vector(0, 0, 1),
+        angle: float = 360,
+        combine: bool = True,
+        clean: bool = True,
+    ) -> T:
+        """
+        Returns a cone with the specified radius and height for each point on the stack
+        A truncated cone can be created by specifying parameters radius1 and radius2 instead of radius.
+        :param height: The height of the cone
+        :type height: float > 0
+        :param radius: The radius of the cone
+        :type radius: float > 0
+        :param radius1: The radius of the bottom of the cone
+        :type radius1: float > 0
+        :param radius2: The radius of the top of the cone
+        :type radius2: float > 0
+        :param direct: The direction axis for the creation of the cone
+        :type direct: A three-tuple
+        :param angle: The angle to sweep the cone arc through
+        :type angle: float > 0
+        :param combine: Whether the results should be combined with other solids on the stack
+            (and each other)
+        :type combine: true to combine shapes, false otherwise
+        :param clean: call :py:meth:`clean` afterwards to have a clean shape
+        :return: A cone object for each point on the stack
+
+        One cone is created for each item on the current stack. If no items are on the stack, one
+        cone is created using the current workplane center.
+
+        If combine is true, the result will be a single object on the stack. If a solid was found
+        in the chain, the result is that solid with all cones produced fused onto it otherwise,
+        the result is the combination of all the produced cones.
+
+        If combine is false, the result will be a list of the cones produced.
+        """
+
+        r1 = radius if radius1 is None or radius1 == 0 else radius1
```

Comment:
```suggestion
        r1 = radius1
```

Reply:
There is a reason for this, it to allow the user to specify `cone(height=40, radius=10) ` for a simple cone and `cone(height=40, radius1=10, radius2=5)` for a truncated cone. This is akin to how OpenSCAD handles things.

Reply:
This is only convenient if argument names are always specified. But what about `cone(40,10)` for simple cone and `cone(40,10,5)` for truncated cone? `cone(40,10,5)` will give surprising results with three radius arguments.

Reply:
True. I could solve that problem by ordering the aguments: `cone(height, radius1, radius2, radius)`. Then
```
cone(40,10)
cone(40,10,5)
cone(height=40,radius=10)
cone(height=40,radius1=10,radius2=5)
```
All behave as expected.

Reply:
The API of OpenSCAD is not really relevant here (different language; no intended relation with CQ). What is very relevant on the other hand is consistence with existing functions. Taking this into account, I'd propose:
```python
def cone(
        self: T,
        radius1: float,
        height: float,
        radius2: float = 0.0,
       ...
```

The same holds for the recently merged cylinder (radius and height should be interchanged to be similar to e.g. the hole signature). It would be really great, if you could amend it in this PR

Reply:
> The API of OpenSCAD is not really relevant here (different language; no intended relation with CQ).

While I agree that there is no need to be the same as OpenSCAD, I don't believe other languages are not relevant - taking cues from how other established systems work can often suggest a pattern to use.

> What is very relevant on the other hand is consistence with existing functions

Agreed. The reason for consistency is not for its own sake, it is because consistency helps avoid errors. The parameter ordering `cone(radius1, height, radius2)` is horrible and error-inducing, thus defeating the reason for being consistent. And if we really want to follow other functions, eg `cboreHole` and `cskHole` then we should use `diameter` rather than `radius`. But we already have `sphere`, `fillet`, `ellipse` and `circle` which use `radius`.  So orthogonality is already broken.

Having said all that, I suggest:
```
def cone(
    self: T,
    height: float,
    radius1: float,
    radius2: float = 0.0,
...
```

This:
1. Uses `radius` and so is consistent with the other geometric primitives.
2. Allows a natural parameter ordering for truncated cone.
3. Does not have the contentious `radius` parameter.